### PR TITLE
inital tests for cmitfb

### PR DESCRIPTION
**Pull request by emord**
https://github.com/dimagi/commcare-hq/pull/11431#issuecomment-215427164

@millerdev I was initally going to try to test this using the case summary, but it looks like it uses djangular and I haven't been able to figure out how to test those calls at the moment. Also this doesn't use DB which is nice.

AdvancedForm test is basically a placeholder til we decide how we want to hashtag that

cc: @NoahCarnahan

### CODE REVIEW

File: corehq/apps/app_manager/tests/test_case_refs_from_fb.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        }
+        with patch.object(PreloadAction, 'wrap') as mock:
+            _update_case_refs_from_form_builder(self.form, ref_json)
+            self.assertFalse(mock.called)
+
+    def testTwoReferences(self):
+        ref_json = {
+            "preload": {
+                "/data/question": "name",
+                "/data/other_question":"close_reason"
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_refs_from_fb.py]: E231 missing whitespace after ':'

File: corehq/apps/app_manager/tests/test_case_refs_from_fb.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            }
+        }
+        _update_case_refs_from_form_builder(self.form, ref_json)
+        self.assertEqual(self.form.actions.load_from_form.to_json(), ref_json)
+
+    def testTwoReferences(self):
+        ref_json = {
+            "preload": {
+                "/data/question": "name",
+                "/data/other_question":"close_reason"
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_case_refs_from_fb.py]: E231 missing whitespace after ':'

File: corehq/apps/app_manager/views/forms.py [hunk 3]
```diff
@@ -628,3 +626,7 @@ def form_casexml(request, domain, form_unique_id):
     if domain != app.domain:
         raise Http404()
     return HttpResponse(form.create_casexml())
+
+def _update_case_refs_from_form_builder(form, reference_json):
```
Comment (dimagimon) [inline corehq/apps/app_manager/views/forms.py]: E302 expected 2 blank lines, found 1

File: corehq/apps/app_manager/tests/test_case_refs_from_fb.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+from corehq.apps.app_manager.models import AdvancedForm, Form, PreloadAction
+from corehq.apps.app_manager.views.forms import _update_case_refs_from_form_builder
+
+
+class FormCaseReferenceTest(SimpleTestCase):
+    def setUp(self):
+        self.form = Form()
+
+    def testOneReference(self):
```
Comment (millerdev) [inline corehq/apps/app_manager/tests/test_case_refs_from_fb.py]: Should use `python_naming_convention`

**Comment by millerdev** [comment]
Test failure is unfortunate: killed because it ran out of time. Luckily the tests in this PR passed:
```
testOneReference (corehq.apps.app_manager.tests.test_case_refs_from_fb.FormCaseReferenceTest) ... ok
testTwoReferences (corehq.apps.app_manager.tests.test_case_refs_from_fb.FormCaseReferenceTest) ... ok
testOneReference (corehq.apps.app_manager.tests.test_case_refs_from_fb.AdvancedFormCaseReferenceTest) ... ok
testTwoReferences (corehq.apps.app_manager.tests.test_case_refs_from_fb.AdvancedFormCaseReferenceTest) ... ok
```

File: corehq/apps/app_manager/tests/test_case_refs_from_fb.py (hunk 4 above)
Comment (emord) [inline corehq/apps/app_manager/tests/test_case_refs_from_fb.py]: https://github.com/dimagi/commcare-hq/pull/11449

**Pull request merged**